### PR TITLE
Count_inversion.cpp

### PR DESCRIPTION
**Pull request by mansikagrawal**
Fixes # 1664

#### Checklist:
- [ Done] 4 space indentation.
- [Done ] Coding conventions are followed.
- [Done ] Input is taken dynamically.
- [Done ] Sample Input / Output is added at the end of file.
- [Done ] Logic Documentation (Comments).
- [Done ] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request: Added file Count_inversion.cpp
- 
- 

#### Languages Used:CPP
- 
- 

#### Files Added: Count_inversion.cpp
- 
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by asha15** [comment]
Specify the issue number next to `Fixes #`

**Comment by mansikagrawal** [comment]
@asha15  Please check I had updated the comment

**Comment by somya-kapoor** [comment]
Please change the description [x] instead of [done].

File: Count_Inversion/Count_inversion.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
+// CPP program to count inversion sort in an array
+// inversion count is in an array such that arr[i] > arr[j]  where  i < j 
+#include<iostream>
+using namespace std;
```
Comment (somya-kapoor) [inline Count_Inversion/Count_inversion.cpp]: Leave a line.

File: Count_Inversion/Count_inversion.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+int mergefn(int arr[] ,int temp[], int left, int mid, int right)
```
Comment (somya-kapoor) [inline Count_Inversion/Count_inversion.cpp]: Add space before the comma, not after.

File: Count_Inversion/Count_inversion.cpp [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+{
+    int i = left;
+    int j = mid;
+    int  k = left;
+    int inv_cnt = 0;
+  while( ( i <= mid-1 ) && ( j <= right ) )
```
Comment (somya-kapoor) [inline Count_Inversion/Count_inversion.cpp]: Remove the space between (( and add a space around -.

File: Count_Inversion/Count_inversion.cpp [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        if( arr[i] <= arr[j] )
+            temp[k++] = arr[i++];
+        else
+        {
+             /* if arr[i] is greater than arr[j], then there are (mid – i) inversions.
+                because left and right subarrays are sorted,
+                so all the remaining elements in left-subarray
+                (arr[i+1], arr[i+2] … arr[mid]) will be greater than arr[j] */
+                temp[k++] = arr[j++];
+                inv_cnt += (mid-i);
```
Comment (somya-kapoor) [inline Count_Inversion/Count_inversion.cpp]: add a space around -.

File: Count_Inversion/Count_inversion.cpp [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        }
+    }
+     /* Copy the remaining elements of left subarray
+         (if there are any) to temp*/
+    while( i <= mid-1 )
+        temp[k++] = arr[i++];
+     /* Copy the remaining elements of right subarray
+        (if there are any) to temp*/
+    while( j <= right )
```
Comment (somya-kapoor) [inline Count_Inversion/Count_inversion.cpp]: Remove space after (

File: Count_Inversion/Count_inversion.cpp [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        temp[k++] = arr[j++];
+         /*Copy back the merged elements to original array*/
+  for( i = left ; i <= right ; i++ )
```
Comment (somya-kapoor) [inline Count_Inversion/Count_inversion.cpp]: Remove space after (.
Add space after comma, not before comma

File: Count_Inversion/Count_inversion.cpp [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    
+    }
+    
+    return inv_cnt;
+}
+int main()
+{
+    // test cases t
+	int t;
+	cin>>t;
```
Comment (somya-kapoor) [inline Count_Inversion/Count_inversion.cpp]: Add space around >> and <<

File: Count_Inversion/Count_inversion.cpp [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+	while(t--){
+    // size of array n
+    int n;
+    cin>>n;
+    // input array arr[]
+    int arr[n];
+    for( int i=0 ; i<n ; i++ )
```
Comment (somya-kapoor) [inline Count_Inversion/Count_inversion.cpp]: Add space around the operator.

**Comment by somya-kapoor** [comment]
@mansikagrawal If you are creating a new PR for the same issue, please close the other existing PR.

**Pull request closed without merging**